src/main/java/frc/robot/subsystems/chooser/factory/RealChooserConstants.java

```diff
@@ -0,0 +1,43 @@
+package frc.robot.subsystems.chooser.factory;
+
+import com.revrobotics.SparkLimitSwitch;
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.constants.IDs;
+import frc.robot.hardware.digitalinput.supplied.SuppliedDigitalInput;
+import frc.robot.hardware.motor.sparkmax.BrushlessSparkMAXMotor;
+import frc.robot.hardware.motor.sparkmax.SparkMaxWrapper;
+import frc.robot.hardware.signal.cansparkmax.SparkMaxAngleSignal;
+import frc.robot.hardware.signal.cansparkmax.SparkMaxDoubleSignal;
+import frc.robot.subsystems.chooser.ChooserStuff;
+import frc.utils.AngleUnit;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+public class RealChooserConstants {
+
+	private final static double DEBOUNCE_TIME_SECONDS = 0.05;
+	private final static Debouncer.DebounceType DEBOUNCE_TYPE = Debouncer.DebounceType.kBoth;
+
+	private final static SparkLimitSwitch.Type REVERSE_LIMIT_SWITCH_TYPE = SparkLimitSwitch.Type.kNormallyOpen;
+
+	public static ChooserStuff generateChooserStuff(String logPath) {
+		SparkMaxWrapper wrapper = new SparkMaxWrapper(IDs.CANSparkMAXIDs.CHOOSER);
+		SysIdRoutine.Config config = new SysIdRoutine.Config();
+		BrushlessSparkMAXMotor motor = new BrushlessSparkMAXMotor(logPath, wrapper, config);
+
+		Supplier<Double> voltage = () -> wrapper.getBusVoltage() * wrapper.getAppliedOutput();
```

Comment:
use get voltage (need to update core remind me)

Reply:
there is no get voltage
or do i need to update something to have it?

Reply:
We will update core and then you will merge core to you and have the function

Reply:
k let me know when u do